src/shortcodes/image-and-text.js

```diff
@@ -1,7 +1,8 @@
 const markdownFilter = require("../filters/markdown.js");
 
 module.exports = (content, image, alt, imagePosition, verticalAlignment) => {
-	return `<div class="wp-block-media-text has-media-on-the-${imagePosition} is-vertically-aligned-${verticalAlignment} is-stacked-on-mobile">
+	const imagePositionClass = imagePosition === "right" ? "has-media-on-the-right " : "";
```

Comment:
If "left" is supported, it's probably better to remove it from the editor component so that we can keep your old code of `has-media-on-the-${imagePosition}`, which gives the flexibility of adding the support of "left" in the futre.

BTW, `is-stacked-on-mobile` class is not styled. Can we take the chance to remove it?